lib/Index/IndexRecord.cpp

```diff
@@ -424,14 +426,15 @@ static void addModuleDependencies(ArrayRef<ImportedModule> imports,
           bool withoutUnitName = true;
           if (FU->getKind() == FileUnitKind::ClangModule) {
             auto clangModUnit = cast<ClangModuleUnit>(LFU);
-            if (!clangModUnit->isSystemModule() || indexSystemModules) {
-              withoutUnitName = false;
+            if ((!clangModUnit->isSystemModule() || indexSystemModules)) {
+              withoutUnitName = !indexClangModules;
```

Comment:
Don't we want this check to be:
```
if ((!clangModUnit->isSystemModule() || indexSystemModules) && indexClangModules) {
...
```
? I'm guessing this is what you had because of the added parentheses though. So probably a reason you changed it?

Reply:
Ah yes, I had changed it so we still fetch the proper module name below (see my old, resolved comments, we were getting `Sub1` instead of `ClangModuleC`). If you think it makes sense, I can change this to give the proper module name for system deps as well (although I'm not sure if they'll ever vary in practice).

Reply:
I think that would make sense, yeah.

Reply:
Or I may have misunderstood, if you are talking about Swift native system deps, then these should be fine, no submodules there.

Reply:
Done